### ISSUE TITLE
CZML Sandcastle examples

### ISSUE THREAD

**Issue by pjcozzi** [AnalyticalGraphicsInc/cesium#3027]
@tiffanylu @adamdavidcole let's use this issue to plan and track the CZML examples you will add to Sandcastle.

To get you started, I created a new [czml-examples branch](https://github.com/AnalyticalGraphicsInc/cesium/compare/master...tiffanylu:czml-examples) in your fork.  It has one new Sandcastle example: [Apps/Sandcastle/gallery/CZML Billboard and Label.html](http://localhost:8080/Apps/Sandcastle/index.html?src=CZML%20Billboard%20and%20Label.html&label=CZML).  To see it in action, run Sandcastle using this branch, and select the example from the new CZML tab at the bottom (the missing "A" in "AGI" is a known Chrome issue, see #2997).

**Resources to help you create a roadmap**:
* [CZML Guide](https://github.com/AnalyticalGraphicsInc/cesium/wiki/CZML-Guide)
   * For the most up-to-date info, see the code itself: each [update function](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/DataSources/CzmlDataSource.js#L1628), e.g., [polygon](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/DataSources/CzmlDataSource.js#L1251)
* [Existing CZML examples](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Apps/SampleData)
* [Cesium API geometries examples](http://cesiumjs.org/Cesium/Apps/Sandcastle/index.html?src=Hello%20World.html&label=Showcases)

Also, I created a `gh-pages` branch that would allow the community to preview the examples by browsing to `tiffanylu.github.io/cesium/`, but this seems to redirect to `http://tiffanylu.com/cesium/`.  I'm sure we will figure it out.

**Comment by tiffanylu** [comment]
@pjcozzi I just registered a new domain name, which is apparently still propagating... should be up within a day or two

**Comment by adamdavidcole** [comment]
Our first step will be to port the geometry examples found in sandcastle to CZML examples. Once we have a better grasp at the development process we will brainstorm more CZML examples including examples with animation and public data usage.

**Comment by tiffanylu** [comment]
* [x] Polygon
* [x] Polyline
* [x] Spheres and Ellipsoids
* [x] Rectangle
* [o] Polygon with shape change and movement over time (in progress)
* [] Point (either plotting satellites or something like cities)
* [] Model
* [] Path using GPS flight data
* [] Satellite path visualization (trailTime)

**Comment by pjcozzi** [comment]
Here's a few more ideas to consider for your list:
* Reference properties: https://cesiumjs.org/2014/07/01/Cesium-version-b30-released/
* Different coordinate systems, e.g., `cartographicDegrees`, `cartesian`
* Different time dynamic approaches, e.g., intervals, sampled
* Wall
* Materials
* Minimal vs. complete examples vs. both

Also, for this set of examples, don't worry about the properties starting with `agi_` that are here: https://github.com/AnalyticalGraphicsInc/cesium/wiki/CZML-Content

**Comment by adamdavidcole** [comment]
Added a couple more examples: 
Ellipses 
Points (still looking for a good source of public data to expand this example)

**Comment by pjcozzi** [comment]
I just realized that we also have this CZML example that _may_ be useful for you: http://cesiumjs.org/Cesium/Build/Apps/CesiumViewer/index.html?source=/Gallery/BAMS_C4ISR.czml&lookAt=Aircraft/GlobalHawk_1a

**Comment by pjcozzi** [comment]
@adamdavidcole can you update your task list: https://github.com/AnalyticalGraphicsInc/cesium/issues/3027#issuecomment-142148226

**Comment by tiffanylu** [comment]
Replied to line note on CZML dynamic polygon - intervals, availability.html 
Aside from the flicker issue (which I'm unsure how to resolve), everything else on my end is ready!

**Comment by tiffanylu** [comment]
@pjcozzi I've merged in master, but the merge conflict warning is still there on the pull request. Is there something else I should also be doing?
And when you get a chance, let me know if you can repro the issue on CZML Polygon - Intervals, Availability.html. Thanks!

**Comment by pjcozzi** [comment]
> I've merged in master, but the merge conflict warning is still there on the pull request. Is there something else I should also be doing?

Perhaps you merged in master from your own repo?  We want to merge master from the main Cesium repo.  I did it in 3468b5dc27174d6b4d0bb245b054a308227ea547.

> let me know if you can repro the issue on CZML Polygon - Intervals, Availability.html? Thanks

See https://github.com/AnalyticalGraphicsInc/cesium/pull/3050#discussion_r41382073

**Comment by pjcozzi** [comment]
Here's a TODO list for closing this issue:

* [ ] Address remaining feedback in #3050 and merge it.
* [ ] Create a new pull request with a new CZML example showing a 3D model using CZML.  A few 3D models are [here](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Apps/SampleData/models).  @mramato do you have an example CZML file referencing a 3D model?
* [ ] Create a pull request for a new CZML example using reference properties (https://cesiumjs.org/2014/07/01/Cesium-version-b30-released/), e.g., to place a billboard and label at the same position (then use pixel offset for the label to offset it).  I forgot to mention this last night, but we have it listed above and it was mentioned on the forum, and I don't think we already created an example for this.

**Comment by tiffanylu** [comment]
@pjcozzi oops my bad, thank you for that! 
As a side note, it looks like none of the Bing Maps imagery (Aerial, Aerial w/ Labels, Roads) is working for me as of this morning. Ditto for my friend's computer as well. Since that's the default imagery provider, this is what navigating to Sandcastle looks like for me:
![image](https://cloud.githubusercontent.com/assets/6036502/10348295/3be017cc-6d06-11e5-8d51-fc5a7cc07155.png)
All other imagery providers work fine for me, and I'm also not seeing any updates on the Bing Maps API site.

**Comment by hpinkos** [comment]
The bing maps imagery is working for me right now, but I have seen it go in and out as well.  Someone in the forum reported it too.

**Comment by pjcozzi** [comment]
@tiffanylu don't worry about Bing - it is not related to our work here.

**Comment by pjcozzi** [comment]
@tiffanylu @adamdavidcole the only thing left before we can close this is the 3D model example in the task list [above]().  We are working on getting you a starter example.

In the meantime, please start on #2814.

**Comment by mramato** [comment]
Whoops, just noticed this issue.  See my comments in #3063

**Comment by pjcozzi** [comment]
@tiffanylu @adamdavidcole in addition to the 3D model example, let's address the comments in #3063 in one or more pull requests.  If you have questions on any of them, please comment in that issue.  Most of them are trivial.

We can push back starting on #3063 a few days.

**Comment by pjcozzi** [comment]
Closing this since we are tracking CZML updates in #3063.
